common/protob/messages-cardano.proto

```diff
@@ -96,13 +102,16 @@ message CardanoPublicKey {
  * @next Failure
  */
 message CardanoSignTx {
-    repeated CardanoTxInputType inputs = 1;     // inputs to be used in transaction
-    repeated CardanoTxOutputType outputs = 2;   // outputs to be used in transaction
-    // optional uint32 transactions_count = 3;  // left as a comment so we know to skip the id 3 in the future
-    optional uint32 protocol_magic = 5;         // network's protocol magic
-    optional uint64 fee = 6;                    // transaction fee - added in shelley
-    optional uint64 ttl = 7;                    // transaction ttl - added in shelley
-    optional uint32 network_id = 8;             // network id - mainnet or testnet
+    repeated CardanoTxInputType inputs = 1;             // inputs to be used in transaction
+    repeated CardanoTxOutputType outputs = 2;           // outputs to be used in transaction
+    // optional uint32 transactions_count = 3;          // left as a comment so we know to skip the id 3 in the future
+    optional uint32 protocol_magic = 5;                 // network's protocol magic
+    optional uint64 fee = 6;                            // transaction fee - added in shelley
+    optional uint64 ttl = 7;                            // transaction ttl - added in shelley
+    optional uint32 network_id = 8;                     // network id - mainnet or testnet
+    repeated CardanoTxCertificateType certificates = 9; // transaction certificates - added in shelley
+    repeated CardanoTxWithdrawalType withdrawals = 10;  // transaction withdrawals - added in shelley
+    optional bytes metadata_hash = 11;                  // transaction metadata hash - added in shelley
```

Comment:
Let's send raw metadata instead of metadata hash.

Reply:
Done [here](https://github.com/trezor/trezor-firmware/pull/1145/commits/1bcf5a81ebf3b0df03b25766cd7f94749233fcbf) - see comment above.